### DISCUSSION TITLE
Dimmer with physical "dumb" switch

### DISCUSSION THREAD

**Discussion by MoonshineSG** [Koenkk/zigbee2mqtt#11668] [Devices]
I have a "Lonsonho TS110E_2gang" (https://www.zigbee2mqtt.io/devices/TS110E_2gang.html#lonsonho-ts110e_2gang) dimmer module that is included in a few scenes, one of them being a "ALL OFF". Once that scene is activated, the brightness is set to 0 and when attempting to turn on the switch, the brightness stays at 0 and the light does not turn on. 

Anyone can tell me if this is a bug, a feature or smth that I am doing wrong somewhere... 

LOGS:

`Debug Received Zigbee message from 'Coffee Table Remote', type 'attributeReport', cluster 'genMultistateInput', data '{"presentValue":1}' from endpoint 1 with groupID null
Info MQTT publish: topic 'zigbee2mqtt/Coffee Table Remote', payload '{"action":"single","battery":100,"click":"single","linkquality":255,"voltage":3025}'
Info MQTT publish: topic 'zigbee2mqtt/Coffee Table Remote', payload '{"action":"","battery":100,"click":null,"linkquality":255,"voltage":3025}'
Info MQTT publish: topic 'zigbee2mqtt/Coffee Table Remote', payload '{"action":null,"battery":100,"click":"","linkquality":255,"voltage":3025}'
Info MQTT publish: topic 'zigbee2mqtt/Coffee Table Remote/action', payload 'single'
Info MQTT publish: topic 'zigbee2mqtt/Coffee Table Remote/click', payload 'single'
Debug Received MQTT message on 'zigbee2mqtt/Main Bedroom/l1/set' with data '{"state": "OFF"}'
Debug Publishing 'set' 'state' to 'Main Bedroom'
Debug Received MQTT message on 'zigbee2mqtt/Main Bedroom/l2/set' with data '{"state": "OFF"}'
Debug Publishing 'set' 'state' to 'Main Bedroom'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom', payload '{"brightness_l1":254,"brightness_l2":254,"color_mode":"xy","linkquality":255,"state_l1":"OFF","state_l2":"ON"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l1', payload '{"brightness":254,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l2', payload '{"brightness":254,"state":"ON"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom', payload '{"brightness_l1":254,"brightness_l2":254,"color_mode":"xy","linkquality":255,"state_l1":"OFF","state_l2":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l1', payload '{"brightness":254,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l2', payload '{"brightness":254,"state":"OFF"}'
Debug Received Zigbee message from 'Main Bedroom', type 'attributeReport', cluster '**genLevelCtrl', data '{"currentLevel":0}'** from endpoint 1 with groupID null
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom', payload '{"brightness_l1":0,"brightness_l2":254,"color_mode":"xy","linkquality":255,"state_l1":"OFF","state_l2":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l1', payload '{"brightness":0,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l2', payload '{"brightness":254,"state":"OFF"}'
Debug Received Zigbee message from 'Main Bedroom', type 'attributeReport', cluster '**genLevelCtrl', data '{"currentLevel":0}'** from endpoint 2 with groupID null
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom', payload '{"brightness_l1":0,"brightness_l2":0,"color_mode":"xy","linkquality":255,"state_l1":"OFF","state_l2":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l1', payload '{"brightness":0,"state":"OFF"}'
Info MQTT publish: topic 'zigbee2mqtt/Main Bedroom/l2', payload '{"brightness":0,"state":"OFF"}'

`

**Comment by MoonshineSG** [comment]
OK.... removing (delete from z2m) and adding the device back restore the expected behaviour...  


the device now reports 

`Debug Received Zigbee message from 'Main Bedroom', type 'attributeReport', cluster 'genOnOff', data '{"onOff":0}' from endpoint 2 with groupID null
`

which maintains the brightness.



Now the problem moved to re-activating a scene after a "all_off", does not actually activate the scene. In the UI it shows that it does, but I believe it does not actually send the command. 



Hopefully this can be fixed.. Somehow..

**Comment by MoonshineSG** [comment]
A bit of testing done and this is what I found. 


I have set 3 scenes:
A - state:on, brightness: 100%
B - state:on, brightness: 20%
OFF - state:off


**mode 1** - normal, freshly added device.

OFF -> Scene A = state:on, brightness: 100% (OK) 
Scene A -> Scene B = state:on, brightness: 20% (OK) 
Scene B -> OFF = state:off, brightness 20% (OK) 
~ so far all normal
OFF -> Scene A = state:on, brightness: 100% (OK) 
Scene A -> OFF = state:off, brightness 100% (OK) 
OFF -> Scene A (OK) = state:on, brightness: 100%  ERROR: reported on UI as ON, but light stays OFF)
    
Using physical switch will turn the light on at the last brightness (as expected)


**mode 2**  - once you set the transition to anyhing, the behavior changes (even if you revert it to the default 0) Only way to go back to above behaviour is to remove the device from z2m and add it again. 

OFF -> Scene A = state:on, brightness: 100% (OK) 
Scene A -> Scene B = state:on, brightness: 20% (OK) 
Scene B -> OFF = state:off, brightness 20% (OK) 
~ so far all identical normal
OFF -> Scene A = state:on, brightness: 100% (OK) 
Scene A -> OFF = state:off, brightness 100% (OK) 
OFF -> Scene A (OK) = state:on, brightness: 100% (OK)

Using physical switch will turn the light status ON but brightness is at 0. ERROR: reported on UI as ON, but light stays OFF)

What I expect is both, togling scenes and using physical switch to work.

Using just physical switch to switch the light ON/OFF does not affect the brightness and works as expected (when starting with light ON and some brightness). 


I have an automation that turn the light to 100% if it's below 5 (absolute val), but thats just a workaround and should not be a final solution. As a test with one light, is ok, bu timagine 20-30 light setup... 
Right now, there's no way to check if the light changes status due to physical switch or automation/UI.... 

I only have this dimmer for testing so I am not sure if this is generic z2m problem or specific to this model or even to this particular device. 

Help. Please.

**Comment by ndbroadbent** [comment]
Hi @MoonshineSG, I also have Lonsonho TS110E dimmer modules (among others), and I'm also trying to get these to work with rocker switches (on/off) instead of the momentary push buttons that they are designed for.

If I turn my wall switch on, the brightness cycles up and down until the switch is turned off, and then it stays at this brightness setting. I have to turn the switch on and off quickly to toggle the state from off to on, or on to off.

Sorry I'm not sure if I understand your post and comments, but have you been able to change this behavior? Did you set something up in zigbee2mqtt or Home Assistant to achieve this? Thanks!

**Reply by MoonshineSG** [reply]
I gave up. I replaced the smart module + dumb switch with a smart rotary switch... 
I was trying to avoid this due to the difference in form factor, but I just couldn't figure a way to solve the strange behaviour...

**Comment by mararn1618** [comment]
Still looking for a solution 🥺, but with ZHA
